src/Template.php

```diff
@@ -80,23 +80,16 @@ public function getParent(array $context): self|TemplateWrapper|false
             return $this->parent;
         }
 
-        try {
-            if (!$parent = $this->doGetParent($context)) {
-                return false;
-            }
-
-            if ($parent instanceof self || $parent instanceof TemplateWrapper) {
-                return $this->parents[$parent->getSourceContext()->getName()] = $parent;
-            }
+        if (!$parent = $this->doGetParent($context)) {
+            return false;
+        }
 
-            if (!isset($this->parents[$parent])) {
-                $this->parents[$parent] = $this->loadTemplate($parent);
-            }
-        } catch (LoaderError $e) {
```

Comment:
This try/catch is not needed as it's already taken care of in `Template::loadTemplate()`.